torch/_export/passes/move_to_device_pass.py

```diff
@@ -0,0 +1,81 @@
+from typing import Dict, Union
+
+import torch
+from torch.export import ExportedProgram
+
+
+def _move_nn_param(
+    v: torch.nn.Parameter, location: Union[torch.device, str, Dict[str, str]]
+) -> torch.nn.Parameter:
+    if isinstance(location, dict):
+        if str(v.device) in location.keys():
+            return torch.nn.Parameter(v.to(location[str(v.device)]))
+        else:
+            raise KeyError(f"Device {str(v.device)} not found in location dict")
+    else:
+        return torch.nn.Parameter(v.to(location))
+
+
+def _move_tensor(
+    v: torch.Tensor, location: Union[torch.device, str, Dict[str, str]]
+) -> torch.Tensor:
+    if isinstance(location, dict):
+        if str(v.device) in location.keys():
+            return v.to(location[str(v.device)])
+        else:
+            raise KeyError(f"Device {str(v.device)} not found in location dict")
+    else:
+        return v.to(location)
+
+
+def _move_kwargs(
+    kwargs: Dict[str, object], location: Union[torch.device, str, Dict[str, str]]
+) -> Dict[str, object]:
+    new_kwargs = kwargs.copy()
+    if isinstance(location, dict):
+        if str(kwargs["device"]) in location.keys():
+            new_kwargs["device"] = location[str(kwargs["device"])]
+        else:
+            raise KeyError(f"Device {str(kwargs['device'])} not found in location dict")
```

Comment:
I don't think we need to error out? I think there are cases where you only want to move specific devices to another device, but keep the other device?